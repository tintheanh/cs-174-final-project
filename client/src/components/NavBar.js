import React, { Component } from 'react'
import { Link, withRouter } from 'react-router-dom'

import { logout } from '../utils/auth';

class NavBar extends Component {
  logOut = () => {
    logout(() => this.props.history.push('/'));
  }

  render() {
    const loginRegLink = (
      <ul>
        <li>
          <Link to="/login">
            Login
          </Link>
        </li>
        <li>
          <Link to="/register">
            Register
          </Link>
        </li>
      </ul>
    )

    const userLink = (
      <ul className="navbar-nav">
        <li>
          <Link to="/profile">
            Profile
          </Link>
        </li>
        <li className="nav-item">
          <a href="" onClick={this.logOut}>
            Logout
          </a>
        </li>
      </ul>
    )

    return (
      <nav>
        <div>
          <ul className="navbar-nav">
            <li className="nav-item">
              <Link to="/">
                Home
              </Link>
            </li>
          </ul>
          {localStorage.usertoken ? userLink : loginRegLink}
        </div>
      </nav>
    )
  }
}

export default withRouter(NavBar)
