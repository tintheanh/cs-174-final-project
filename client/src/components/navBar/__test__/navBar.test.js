import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../navBar';

afterEach(cleanup);

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(
		<Router>
			<NavBar />
		</Router>,
		div
	);
});

describe('Welcome (Snapshot)', () => {
	it('Welcome renders hello world', () => {
    // localStorage.setItem("usertoken", "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwidXNlcm5hbWUiOiJKb2huIiwiaWF0IjoxNTE2MjM5MDIyfQ.HaAB9LmgyhESJHqVLTWpd7hejlCxT6dc-sGk9DUlE64")
    
		const component = renderer.create(
			<Router>
				<NavBar />
			</Router>
		);
    const json = component.toJSON();
    
		expect(json).toMatchSnapshot();
	});
});
