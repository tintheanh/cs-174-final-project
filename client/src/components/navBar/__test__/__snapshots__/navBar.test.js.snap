// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome (Snapshot) Welcome renders hello world 1`] = `
<nav
  data-testid="navBar"
>
  <div>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
